monash-human-power/raspicam
# Publish whether the client is connected to the mqtt broker.

## Description

As requested by @khanguslee some number of weeks back, this allows the dashboard to determine when the camera is connected to mqtt (as opposed to the video feed being on). 

The main reason I separated it into a distinct topic (instead of just reusing the recording or video feed topics) is so it's easier to tell when the client was disconnected (through last will) rather than some other disconnect.

For now, I've just made it publish on connect and disconnect, but if we want it to do periodic updates, then I think there should be a bit of refactoring to clean up the `Backend` parameters and the repeated code.  I have a card on Notion which describes this - "Dealing with retained and/or periodic mqtt messages between raspicam and dashboard".

Related to [dashboard#78](https://github.com/monash-human-power/dashboard/pull/78) and [common#9](https://github.com/monash-human-power/common/pull/9).

## Screenshots

![image](https://user-images.githubusercontent.com/50545432/92302649-278ca500-efb1-11ea-8168-21dce376bae0.png)

## Steps to Test

1. Start `mosquitto`

2. Start `raspicam` using `overlay_new.py` (or any other overlay)

3. Observe the `/v3/status/camera` topic
